### PR TITLE
update K400 datatools to support preparation of K600 and K700

### PR DESCRIPTION
**Pull request by kennymckormick**
Extend Kinetics-400 data tools to support data preparation of K600 and K700.

### CODE REVIEW

**Comment by dreamerlin** [comment]
Need to update changelog : )

**Comment by kennymckormick** [comment]
> Need to update changelog and README.md : )

In my last commit, I update the README file to show that we support the 3 versions of Kinetics datasets. But I'm not sure whether it's ur meaning. Is there anything else that needs to be updated?

**Review by dreamerlin — APPROVED** [review]

**Comment by kennymckormick** [comment]
@innerlee , seems OK to merge now.

**Comment by innerlee** [comment]
will take a look in the evening

File: docs/changelog.md [hunk 1]
```diff
@@ -4,6 +4,7 @@
 
 **Improvements**
 - Set default values of 'average_clips' in each config file so that there is no need to set it explicitly during testing in most cases ([#232](https://github.com/open-mmlab/mmaction2/pull/232))
+- Modify Kinetics data tools to support the preparation of Kinetics-600 and Kinetics-700 too ([#254](https://github.com/open-mmlab/mmaction2/pull/254))
```
Comment (innerlee) [inline docs/changelog.md:7]: `Support data preparation for Kinetics-600 and Kinetics-700`

File: docs/supported_datasets.md [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 - [UCF101](https://www.crcv.ucf.edu/research/data-sets/ucf101/): See [preparing_ucf101](/tools/data/ucf101/preparing_ucf101.md).
 - [HMDB51](https://serre-lab.clps.brown.edu/resource/hmdb-a-large-human-motion-database/): See [preparing_hmdb51](/tools/data/hmdb51/preparing_hmdb51.md).
-- [Kinetics400](https://deepmind.com/research/open-source/kinetics): See [preparing_kinetics400](/tools/data/kinetics400/preparing_kinetics400.md)
+- [Kinetics](https://deepmind.com/research/open-source/kinetics): See [preparing_kinetics](/tools/data/kinetics/preparing_kinetics.md)
```
Comment (innerlee) [inline docs/supported_datasets.md:5]: `[Kinetics-[400/600/700]](...)`

File: tools/data/kinetics/preparing_kinetics.md [hunk 3]
```diff
@@ -1,14 +1,14 @@
-# Preparing Kinetics-400
+# Preparing Kinetics
```
Comment (innerlee) [inline tools/data/kinetics/preparing_kinetics.md:1]: `Preparing  Kinetics-[400/600/700]`

File: tools/data/kinetics/extract_frames.sh [hunk 4]
```diff
@@ -0,0 +1,12 @@
+#!/usr/bin/env bash
+
+DATASET=$1
+
+cd ../
+python build_rawframes.py ../../data/${DATASET}/videos_train/ ../../data/${DATASET}/rawframes_train/ --level 2 --flow-type tvl1 --ext mp4 --task both  --new-width 340 --new-height 256
```
Comment (innerlee) [inline tools/data/kinetics/extract_frames.sh:6]: add a note in preparing_kinetics.md, to remind users that it is using the fixed resolution of 340x256.
users can change the script if they want other options.

File: tools/data/kinetics/generate_rawframes_filelist.sh [hunk 5]
```diff
@@ -0,0 +1,11 @@
+#!/usr/bin/env bash
+
+DATASET=$1
```
Comment (innerlee) [inline tools/data/kinetics/generate_rawframes_filelist.sh:3]: whether we should guard against bad arguments, something like (not tested)
 
```bash
if [ $# -eq 0 ]; then
    echo "No arguments provided"
    exit 1
fi
```

File: tools/data/kinetics/extract_frames.sh (hunk 4 above)
Comment (kennymckormick) [reply tools/data/kinetics/extract_frames.sh:6]: There already is a note in preparing_kinetics.md, at the end of Step 3.

File: tools/data/kinetics/generate_rawframes_filelist.sh (hunk 5 above)
Comment (kennymckormick) [reply tools/data/kinetics/generate_rawframes_filelist.sh:3]: I plan to use something like the following script to test the validity of the argument. [Maybe ugly]

```
DATASET=$1
if [ "$DATASET" == "kinetics400" ] || [ "$1" == "kinetics600" ] || [ "$1" == "kinetics700" ]; then
        echo "We are processing $DATASET"
else
        echo "Bad argument"
        exit 0
fi
```

**Pull request merged**